File: addin-example-powermill-plugin/ExamplePowerMillPlugin/ExamplePowerMillPlugin.cs
// -----------------------------------------------------------------------
// Copyright 2021 Autodesk, Inc. All rights reserved.
//
// Use of this software is subject to the terms of the Autodesk license
// agreement provided at the time of installation or download, or which
// otherwise accompanies this software in either electronic or hard copy form.
// -----------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;
using Delcam.Plugins.Framework;

namespace ExamplePowerMillPlugin
{
//TODO: "Change this Guid to be something unique";
    [Guid("0D229F75-EA25-4B51-8EA5-991232FFCCE6")]
    [ClassInterface(ClassInterfaceType.None)]
    [ComVisible(true)]
    public class ExamplePowerMillPlugin : PluginFrameworkWithPanesAndTabs
    {
        public override string PluginAssemblyName => "ExamplePowerMillPlugin";
//TODO: "Change this Guid to match the one above that you uniquely generated";
        public override Guid PluginGuid => new Guid("0D229F75-EA25-4B51-8EA5-991232FFCCE6");
        public override string PluginName => "ExamplePowerMillPlugin";
        public override string PluginAuthor => "Your name here";
        public override string PluginDescription => "Example plugin";
        public override string PluginIconPath => null;
        public override Version PluginVersion => new Version(1, 0);
        public override Version PowerMILLVersion => new Version(2021, 0);
        public override bool PluginHasOptions => false;
        protected override void register_panes()
        {
            var pane = new VerticalPaneExample();
            register_pane(new PaneDefinition(pane, 900, 375, "Plugin for PowerMill", null));

        }

        protected override void register_tabs()
        {
            var tab = new HorizontalTabExample();
            register_tab(new TabDefinition(tab, 200, "Plugin for PowerMill", null));
        }
    }
}

File: addin-example-powermill-plugin/ExamplePowerMillPluginVB/ExamplePowerMillPlugin.vb
' -----------------------------------------------------------------------
' Copyright 2021 Autodesk, Inc. All rights reserved.
'
' Use of this software is subject to the terms of the Autodesk license
' agreement provided at the time of installation or download, or which
' otherwise accompanies this software in either electronic or hard copy form.
' -----------------------------------------------------------------------

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.InteropServices
Imports System.Text
Imports System.Threading.Tasks
Imports Delcam.Plugins.Framework

<Guid("0D229F75-EA25-4B51-8EA5-991232FFCCE6")>
<ClassInterface(ClassInterfaceType.None)>
<ComVisible(true)>
Public Class ExamplePowerMillPlugin
    Inherits PluginFrameworkWithPanesAndTabs

    Public Overrides ReadOnly Property PluginAssemblyName As String
        Get
            Return "ExamplePowerMillPlugin"
        End Get
    End Property

    Public Overrides ReadOnly Property PluginGuid As Guid
        Get
            Return New Guid("0D229F75-EA25-4B51-8EA5-991232FFCCE6")
        End Get
    End Property

    Public Overrides ReadOnly Property PluginName As String
        Get
            Return "ExamplePowerMillPlugin"
        End Get
    End Property

    Public Overrides ReadOnly Property PluginAuthor As String
        Get
            Return "Your name here"
        End Get
    End Property

    Public Overrides ReadOnly Property PluginDescription As String
        Get
            Return "Example plugin"
        End Get
    End Property

    Public Overrides ReadOnly Property PluginIconPath As String
        Get
            Return Nothing
        End Get
    End Property

    Public Overrides ReadOnly Property PluginVersion As Version
        Get
            Return New Version(1, 0)
        End Get
    End Property

    Public Overrides ReadOnly Property PowerMILLVersion As Version
        Get
            Return New Version(2021, 0)
        End Get
    End Property

    Public Overrides ReadOnly Property PluginHasOptions As Boolean
        Get
            Return False
        End Get
    End Property

    Protected Overrides Sub register_panes()
        Dim pane = New VerticalPaneExample()
        register_pane(New PaneDefinition(pane, 900, 375, "Plugin for PowerMill", Nothing))
    End Sub

    Protected Overrides Sub register_tabs()
        Dim tab = new HorizontalTabExample()
        register_tab(new TabDefinition(tab, 200, "Plugin for PowerMill", Nothing))
    End Sub

End Class
